From c8af956ed26bc8f18b6981f55dcfc40c14d188cf Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:17 -0400
Subject: [scsi] qla2xxx: Remove init control block related dead code for ISPFX00

Message-id: <1408391817-6389-30-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87945
O-Subject: [PATCH RHEL 7.1 e-stor 29/69] qla2xxx: Remove init control block related dead code for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Saurav Kashyap <saurav.kashyap@qlogic.com>

scsi commit id 322316189d42e6ba2c163af1e0f882fc4eb055d6

Signed-off-by: Giridhar Malavali <giridhar.malavali@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_def.h b/drivers/scsi/qla2xxx/qla_def.h
index f9c79a1..b910edf 100644
--- a/drivers/scsi/qla2xxx/qla_def.h
+++ b/drivers/scsi/qla2xxx/qla_def.h
@@ -1197,30 +1197,6 @@ typedef struct {
  uint8_t  reserved_3[26];
 } init_cb_t;
 
-
-struct init_cb_fx {
- uint16_t version;
- uint16_t reserved_1[13];
- __le16  request_q_outpointer;
- __le16  response_q_inpointer;
- uint16_t reserved_2[2];
- __le16  response_q_length;
- __le16  request_q_length;
- uint16_t reserved_3[2];
- __le32  request_q_address[2];
- __le32  response_q_address[2];
- uint16_t reserved_4[4];
- uint8_t  response_q_msivec;
- uint8_t  reserved_5[19];
- uint16_t interrupt_delay_timer;
- uint16_t reserved_6;
- uint32_t fwoptions1;
- uint32_t fwoptions2;
- uint32_t fwoptions3;
- uint8_t  reserved_7[24];
-};
-
-
 /*
  * Get Link Status mailbox command return buffer.
  */
diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index f897d05..1f1f112 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -631,20 +631,6 @@ qlafx00_config_rings(struct scsi_qla_host *vha)
 {
  struct qla_hw_data *ha = vha->hw;
  struct device_reg_fx00 __iomem *reg = &ha->iobase->ispfx00;
- struct init_cb_fx *icb;
- struct req_que *req = ha->req_q_map[0];
- struct rsp_que *rsp = ha->rsp_q_map[0];
-
- /* Setup ring parameters in initialization control block. */
- icb = (struct init_cb_fx *)ha->init_cb;
- icb->request_q_outpointer = __constant_cpu_to_le16(0);
- icb->response_q_inpointer = __constant_cpu_to_le16(0);
- icb->request_q_length = cpu_to_le16(req->length);
- icb->response_q_length = cpu_to_le16(rsp->length);
- icb->request_q_address[0] = cpu_to_le32(LSD(req->dma));
- icb->request_q_address[1] = cpu_to_le32(MSD(req->dma));
- icb->response_q_address[0] = cpu_to_le32(LSD(rsp->dma));
- icb->response_q_address[1] = cpu_to_le32(MSD(rsp->dma));
 
  WRT_REG_DWORD(&reg->req_q_in, 0);
  WRT_REG_DWORD(&reg->req_q_out, 0);
diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 9767fc9..fa0c3d6 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2505,7 +2505,6 @@ qla2x00_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
   ha->aen_mbx_count = AEN_MAILBOX_REGISTER_COUNT_FX00;
   req_length = REQUEST_ENTRY_CNT_FX00;
   rsp_length = RESPONSE_ENTRY_CNT_FX00;
-  ha->init_cb_size = sizeof(struct init_cb_fx);
   ha->isp_ops = &qlafx00_isp_ops;
   ha->port_down_retry_count = 30; /* default value */
   ha->mr.fw_hbt_cnt = QLAFX00_HEARTBEAT_INTERVAL;
-- 
1.7.1